# Data
We are going to use the data from the following link:
https://www.kaggle.com/datasets/asaniczka/1-3m-linkedin-jobs-and-skills-2024

In [5]:
import numpy as np # linear algebra
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [11]:
# Load the data
job_skills = pd.read_csv('E:/Programming/cu-boulder/machine-learning/Course 1/project/data/job_skills.csv')
job_summaries = pd.read_csv('E:/Programming/cu-boulder/machine-learning/Course 1/project/data/job_summary.csv')
job_postings = pd.read_csv('E:/Programming/cu-boulder/machine-learning/Course 1/project/data/linkedin_job_postings.csv')

In [19]:
print("Job Skills Invalid Values")
print(job_skills.isna().sum())
print("")
print("Job Summaries Invalid Values")
print(job_summaries.isna().sum())
print("")
print("Job Postings Invalid Values")
print(job_postings.isna().sum())

Job Skills Invalid Values
job_link         0
job_skills    2085
dtype: int64

Job Summaries Invalid Values
job_link       0
job_summary    0
dtype: int64

Job Postings Invalid Values
job_link                0
last_processed_time     0
got_summary             0
got_ner                 0
is_being_worked         0
job_title               0
company                11
job_location           19
first_seen              0
search_city             0
search_country          0
search_position         0
job_level               0
job_type                0
dtype: int64


As we can see, most of our data looks quite good.
We will drop the rows with missing values in the job_skills dataset, because they are unuseable.
We can work with the (very few) missing company and job_location values in the job_postings dataset though.

In [20]:
# Drop the rows with missing values
job_skills = job_skills.dropna()

Let's have a look at the different job titles and search positions in the job_postings dataset.

In [28]:
unique_titles = job_postings['job_title'].str.lower().unique()
print("Unique Job Titles: ", len(unique_titles))
print("Unique Job Titles (lower case): ", len(unique_titles))
print(job_postings['job_title'].str.lower().value_counts())
print("\n")

unique_positions = job_postings['search_position'].unique()
print("Unique Search Positions: ", len(unique_positions))
print(job_postings['search_position'].value_counts())

Unique Job Titles:  579066
job_title
customer service representative                                                                                                            9436
lead sales associate-ft                                                                                                                    7325
shift manager                                                                                                                              5849
store manager                                                                                                                              5748
assistant manager                                                                                                                          5540
                                                                                                                                           ... 
clinical rn sharp memorial 1w oncology ft night; currently offering up to $10,000 for new hires to 

As we can see, we have many more unique job titles than search positions (584544 Job Titles vs. 1993 Search Positions).
If we convert the job titles to lower case, we drop by about 5000 to 579066 unique job titles. This also changes the job titles that occur most often (apparently, "Customer Service Representative" was written in different ways and accounted for 4000 more listings when converted to lower case).
Lowercasing search positions does not lead to a change.

It now seems like the "search position" would be a better job indicator than the job title, because the title frequently includes additional info, such as the location ("Survey Crew Chief - Michigan"), the field ("Quality Engineer - Aerospace"), the job level ("Designer 1") or general info ("Employment Attorney with relevant experience"); unfortunately, search position and job title frequently seem to be unrelated, as the examples below show.

In [43]:
print(job_postings.loc[56]['job_title'], " - ", job_postings.loc[56]['search_position'])
print(job_postings.loc[58]['job_title'], " - ", job_postings.loc[58]['search_position'])
print(job_postings.loc[64]['job_title'], " - ", job_postings.loc[64]['search_position'])
print(job_postings.loc[91]['job_title'], " - ", job_postings.loc[91]['search_position'])
print(job_postings.loc[124]['job_title'], " - ", job_postings.loc[124]['search_position'])
print(job_postings.loc[164]['job_title'], " - ", job_postings.loc[164]['search_position'])
print(job_postings.loc[177]['job_title'], " - ", job_postings.loc[177]['search_position'])

Designer I  -  Tattoo Artist
Regional Sales Manager, Mid-Atlantic  -  Director Of Religious Activities
Surgical Sales Representative  -  Handle Maker
Keyholder  -  Reader
Lead Software Engineer, Back End  -  Job Development Specialist
RN - Registered Nurse - 24060503EXPPLAT  -  Driver
Android Engineer  -  Tool Grinder Ii
